internal/xds/matcher/matcher_header_test.go

```diff
@@ -467,3 +467,53 @@ func TestHeaderSuffixMatcherMatch(t *testing.T) {
 		})
 	}
 }
+
+func TestHeaderStringMatch(t *testing.T) {
+	tests := []struct {
+		name   string
+		key    string
+		sm     StringMatcher
+		md     metadata.MD
+		invert bool
```

Comment:
Super nit: please swap these two in order (here and in the test cases) since `invert` is part of the matcher and `md` is the input passed to the matcher.  OR: build the matcher in the testcase list and put it in there instead of the individual fields.

Reply:
Done. Chose the first option and swapped.